xline/src/storage/execute_error.rs

```diff
@@ -1,134 +1,103 @@
-    }
-
+    #[error("lease ttl is too large: {0}")]
+    LeaseTtlTooLarge(i64),
     /// Lease already exists
-    pub(crate) fn lease_already_exists(lease_id: i64) -> Self {
-        Self::LeaseError(format!("lease {lease_id} already exists"))
-    }
+    #[error("lease {0} already exists")]
+    LeaseAlreadyExists(i64),
 
+    // AuthErrors
     /// Auth is not enabled
-    pub(crate) fn auth_not_enabled() -> Self {
-        Self::AuthError("auth is not enabled".to_owned())
-    }
-
+    #[error("auth is not enabled")]
+    AuthNotEnabled,
     /// Auth failed
-    pub(crate) fn auth_failed() -> Self {
-        Self::AuthError("invalid username or password".to_owned())
-    }
-
+    #[error("invalid username or password")]
+    AuthFailed,
     /// User not found
-    pub(crate) fn user_not_found(username: &str) -> Self {
-        Self::AuthError(format!("user {username} not found"))
-    }
-
+    #[error("user {0} not found")]
+    UserNotFound(String),
     /// User already exists
-    pub(crate) fn user_already_exists(username: &str) -> Self {
-        Self::AuthError(format!("user {username} already exists"))
-    }
-
+    #[error("user {0} already exists")]
+    UserAlreadyExists(String),
     /// User already has role
-    pub(crate) fn user_already_has_role(username: &str, rolename: &str) -> Self {
-        Self::AuthError(format!("user {username} already has role {rolename}"))
-    }
-
-    /// Permission already exists
-    pub(crate) fn no_password_user() -> Self {
-        Self::AuthError("password was given for no password user".to_owned())
-    }
-
+    #[error("user {0} already has role {1}")]
+    UserAlreadyHasRole(String, String),
+    /// Password was given for no password user
+    #[error("password was given for no password user")]
+    NoPasswordUser,
     /// Role not found
-    pub(crate) fn role_not_found(rolename: &str) -> Self {
-        Self::AuthError(format!("role {rolename} not found"))
-    }
-
+    #[error("role {0} not found")]
+    RoleNotFound(String),
     /// Role already exists
-    pub(crate) fn role_already_exists(rolename: &str) -> Self {
-        Self::AuthError(format!("role {rolename} already exists"))
-    }
-
+    #[error("role {0} already exists")]
+    RoleAlreadyExists(String),
     /// Role not granted
-    pub(crate) fn role_not_granted(rolename: &str) -> Self {
-        Self::AuthError(format!("role {rolename} is not granted to the user"))
-    }
+    #[error("role {0} is not granted to the user")]
+    RoleNotGranted(String),
     /// Root role not exist
-    pub(crate) fn root_role_not_exist() -> Self {
-        Self::AuthError("root user does not have root role".to_owned())
-    }
-
+    #[error("root user does not have root role")]
+    RootRoleNotExist,
     /// Permission not granted
-    pub(crate) fn permission_not_granted() -> Self {
-        Self::AuthError("permission not granted to the role".to_owned())
-    }
-
+    #[error("permission not granted to the role")]
```

Comment:
What are the differences between `PermissionNotGranted` and `PermissionNotGiven`?